frontend/src/app/modules/availability/new-slot/new-availability-page/new-availability-page.component.ts

```diff
@@ -1,16 +1,91 @@
     templateUrl: './new-availability-page.component.html',
     styleUrls: ['./new-availability-page.component.sass'],
 })
 export class NewAvailabilityPageComponent {
-    // eslint-disable-next-line no-empty-function
-    constructor(private router: Router) { }
+    constructor(
+        private router: Router,
+        private slotService: AvailabilitySlotService,
+        private notifications: NotificationService,
+        // eslint-disable-next-line no-empty-function
+    ) { }
+
+    public currentUserId: number = 2;
+
+    private unsubscribe$ = new Subject<void>();
+
+    @ViewChild(NewAvailabilityComponent) newAvailabilityComponent: NewAvailabilityComponent;
 
     public goToPage(pageName: string) {
         this.router.navigate([`${pageName}`]);
     }
+
+    public saveChanges() {
+        const newSlot = this.getNewAvailability();
+
+        this.slotService.createSlot(newSlot)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe(
+                () => {
+                    this.notifications.showSuccessMessage('Slot was successfully updated');
+                    this.goToPage('/availability');
+                },
+                (error) => {
+                    this.notifications.showErrorMessage(error);
+                },
+            );
+    }
+
+    private getNewAvailability() {
+        const general = this.newAvailabilityComponent.generalComponent.settings;
+        const eventDetails = this.newAvailabilityComponent.eventDetailComponent.settings;
+        const advancedSettings = this.newAvailabilityComponent.generalComponent.addAdvanced
+            ? {
+                activityType: general.slotActivityOption,
+                days: general.slotSize,
+                maxNumberOfBookings: general.maxBookings,
+                paddingMeeting: general.meetingPadding,
+                minBookingMeetingDifference: general.minBookingMeetingDifference,
+                color: general.color,
+            } : null;
+        const newAvailability: INewAvailability = {
+            generalDetails: {
+                locationType: general.meetingLocation,
+                name: general.meetingName,
+                type: SlotType.Personal,
+                size: general.slotSize,
+                frequency: general.slotFrequency,
+                isEnabled: this.newAvailabilityComponent.isActive,
+                isVisible: general.hideFromCommon,
+            },
+            eventDetails: {
+                link: eventDetails.linkChoice,
+                welcomeMessage: eventDetails.welcomeMessage,
+                language: eventDetails.languageSelect,
+                bookingsPerDay: eventDetails.allowBookingSelect,
+                allowToAddGuests: eventDetails.isAllowBooker,
+                passwordProtectionIsUsed: eventDetails.passwordProtect,
+                passwordProtection: eventDetails.passwordInput,
+                timeZoneVisibility: eventDetails.zoneChoice,
```

Comment:
Why do you need this mapping?

Reply:
Frontend interfaces don't match backend dtos in terms of property names and changing them would require changing update availability dtos and their mapping as well